### PR TITLE
Restructured app, updated code review observations

### PR DESCRIPTION
**Pull request by lisandro-montano**
Still pending changes in list_view:
- sub levels allowed
- some documentation updates
  Still pending changes in file_properties:
- documentation and code review corrections

Both will be sent by Gaby

### CODE REVIEW

**Comment by lisandro-montano** [comment]
@deadlykyo Hi Edson,

I have applied most of code review comments in this branch and want to merge it with develop.

Please review and consider that there are still code review changes pending explained in my previous comment in this pull request. Those changes will be sent by Gaby later today.

**Comment by gabriela-becerra-jala** [comment]
@deadlykyo Hello Edson,

The missing classes (ListView and ItemOperations) have now applied the code fixes according to your comments.
Please, let us know if something else is missing.

Thank you in advance.

**Comment by lisandro-montano** [comment]
@deadlykyo I have fixed a problem with tab key that went through action bar too, now, it only woks on panels

File: items_management/base_item.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+from PyQt4 import QtGui
+from PyQt4 import QtCore
+
+class BaseItem(QtGui.QFileSystemModel):
+	def __init__(self, current_path):
+		"""Create BaseItem to handle all item_operations"""
```
Comment (deadlykyo) [inline items_management/base_item.py]: The parameter of the constructor is not documented

File: items_management/item_operations.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+class ItemOperations(object):
+    def __init__(self):
+        super(ItemOperations, self).__init__()
+
+    def item_data(self, panel, index, data):
+        """Returns the item information according to the required data requested.
+
+        Params:
+        - panel: list_view object
```
Comment (deadlykyo) [inline items_management/item_operations.py]: remember to put examples of the expected values as parameters or return values

File: items_management/item_operations.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        - index: selected item index
+        - data: item data type requested
+        """
+        index_item = panel.model().index(index.row(), 0, index.parent())
```
Comment (deadlykyo) [inline items_management/item_operations.py]: what it means 0? , avoid the usage of hardcoded values

File: items_management/item_operations.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+        if (data == "Name"):
+            return panel.model().fileName(index_item)
+
+        elif data == "Path":
```
Comment (deadlykyo) [inline items_management/item_operations.py]: be consistent with your coding style, the first if has parenthesis but the remaining does not have

File: items_management/item_operations.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            return panel.model().filePath(index_item)
+
+        elif data == "Info":
+            return panel.model().fileInfo(index_item)
+
+        elif data == "Type":
+            return panel.model().type(index_item)
+
+    def get_item_type(self, panel, index):
+        """Returns 1 if the item is File and returns 0 if the item is Folder
```
Comment (deadlykyo) [inline items_management/item_operations.py]: The comments are not correct, this method does not return 1 or 0

File: items_management/item_operations.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        - panel: list_view object
+        - index: selected item index
+        """
+        file_type = self.item_data(panel, index, "Type")
+        if self.sub_string(str(file_type), "left", -6).find('Folder') >= 0:
+            return 'Folder'
+        elif self.sub_string(str(file_type), "left", -4).find("File") >= 0:
+            return "File"
+
+    def rename_dialog(self, panel, index):
```
Comment (deadlykyo) [inline items_management/item_operations.py]: it is important keep separated the layers of the UI and the logic of the application, in this method we have UI and logic code

File: items_management/item_operations.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+        #If the
+        if ok and current_file_name != new_name:
+            os.rename(current_file_path, self.new_item_path(current_file_path, current_file_name, new_name))
+
+    def new_item_path(self, current_path, name, new_name):
+        """Returns the new item path.
+
+        Params:
+        - panel: list_view object
```
Comment (deadlykyo) [inline items_management/item_operations.py:17]: panel is not a parameter

File: items_management/item_operations.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+        new_name, ok = QtGui.QInputDialog.getText(panel, 'Rename %s Dialog' % current_type,
+                                              'Modify the %s name:' % current_type,
+                                              QtGui.QLineEdit.Normal, current_file_name)
+
+        #If the
+        if ok and current_file_name != new_name:
+            os.rename(current_file_path, self.new_item_path(current_file_path, current_file_name, new_name))
+
+    def new_item_path(self, current_path, name, new_name):
+        """Returns the new item path.
```
Comment (deadlykyo) [inline items_management/item_operations.py:14]: new_item_path does not make reference to an action remember that methods should always be verbs or actions, and the description of the method is too general, how is the new path, why do i need to use this method, what is the special functionality that provides

File: items_management/item_operations.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        - panel: list_view object
+        - current_path: the current complete item path
+        - name: the older item name
+        - new_name: the new item name
+        """
+        path = self.sub_string(current_path, "right", -len(name))
+        return path + new_name
+
+    def sub_string(self, string, side, quantity):
+        """Returns a substring according to the side and letter quantity send.
```
Comment (deadlykyo) [inline items_management/item_operations.py:31]: this method is still too general, is hard to understand what it does reading the documentation an example will help to understand the purpose of this method

File: items_management/item_operations.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+        Params:
+        - string: the string from where the substring will be obtained
+        - side: the string side from where the position counter will start
+        - quantity: number of sites the position counter will move to obtain the substring
+        """
+        if side == "left":
```
Comment (deadlykyo) [inline items_management/item_operations.py]: remember not to use hardcoded values like "left" and "right" use constants

File: views/menus_toolbars/panel_toolbar.py [hunk 11]
```diff
@@ -1,10 +1,12 @@
 from PyQt4 import QtGui
 from PyQt4 import QtCore
 
-
 class PanelToolbar(QtGui.QWidget):
 	def __init__(self, current_path):
-		"""Create panel toolbar"""
+		"""Create panel toolbar including:
+		- Combo box to list computer drives depending on the OS
```
Comment (deadlykyo) [inline views/menus_toolbars/panel_toolbar.py:7]: the documentation does not help too mucha and there is no description for the parameter

Comment (deadlykyo) [inline views/menus_toolbars/panel_toolbar.py:7]: This method has more than 20 lines of code

File: views/menus_toolbars/panel_toolbar.py [hunk 12]
```diff
@@ -33,15 +40,22 @@ def get_volume_list(self):
...
 	def attach(self, observer):
-		"""Attach observers to detect directory changes"""
+		"""Attach observers to detect directory/path changes"""
 		if not observer in self._observers:
 			self._observers.append(observer)
 
-	def propagate_dir(self, new_dir):
-		"""Inform observers about changes in directory path"""
+	def propagate_dir(self, new_dir = ""):
+		"""Inform observers about changes in directory path
```
Comment (deadlykyo) [inline views/menus_toolbars/panel_toolbar.py:56]: documentation for the parameter of the method, please review this across all methods

File: views/panels/panel_view.py [hunk 13]
```diff
@@ -29,18 +33,25 @@ def __init__(self):
...
+		elif type == self.ICONS_VIEW:
 			self.panel = IconsView(self.current_path)
-		elif type == 3:
+		elif type == self.DETAILED_VIEW:
 			self.panel = DetailsView(self.current_path)
 		return self.panel
 
 	def propagate_dir(self, new_dir):
-		"""Detect changes in directory and propagate them to proper instances"""
+		"""Detect changes in directory/path and propagate them to proper instances
```
Comment (deadlykyo) [inline views/panels/panel_view.py:50]: documentation for parameters of the methods

File: views/view_styles/list_view.py [hunk 14]
```diff
@@ -0,0 +1,148 @@
...
+from PyQt4 import QtGui
+from PyQt4 import QtCore
+from items_management.item_operations import ItemOperations
+from items_management.base_item import BaseItem
+from PyQt4.QtGui import QAbstractItemView, QTableView
+from PyQt4.QtCore import Qt
+
+class ListView(QtGui.QTableView):
+	def __init__(self, current_path):
+		"""Sets the current path items view as list"""
```
Comment (deadlykyo) [inline views/view_styles/list_view.py]: no documentation for the parameters

File: views/view_styles/list_view.py [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+		super(ListView, self).__init__()
+		self.item_operations=ItemOperations()
```
Comment (deadlykyo) [inline views/view_styles/list_view.py]: spaces after and before the assignation

File: views/view_styles/list_view.py [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+		self.setGridStyle(Qt.NoPen)
+
+		#Hiding the not required columns
+		self.setColumnWidth(NAME_COLUMN, NAME_COLUMN_WIDTH)
+		self.hideColumn(SIZE_COLUMN)
+		self.hideColumn(DATE_COLUMN)
+		self.hideColumn(KIND_COLUMN)
+
+	def update_path(self, new_path):
+		"""Update panel root index to modify after path changes
```
Comment (deadlykyo) [inline views/view_styles/list_view.py:49]: no docs for the parameter

**Comment by deadlykyo** [comment]
Please review the comments inline and we can do a quick check tomorrow in the monitoring before the merge of the code

**Comment by gabriela-becerra-jala** [comment]
@deadlykyo Hello Edson,

Almost all the observations you did were fixed. Is still missing the changes related to panel_toolbar and BaseItem update_path methods. Lisandro will add the fixes before noon.
Please review the commit I did, and let me know if something else is incorrect.

Thank you in advance,
Gaby

**Comment by lisandro-montano** [comment]
Merging to develop, there we can check the updates prior to performing a merge to master. This way, we can continue working from this updated branch

**Pull request merged**